src/theory/partition_generator.h

```diff
@@ -0,0 +1,128 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Amalee Wilson, Andrew Wu
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * PartitionGenerator for creating partitions.
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__THEORY__SPLITTER_H
+#define CVC5__THEORY__SPLITTER_H
+
+#include <fstream>
+#include <sstream>
+#include <vector>
+
+#include "proof/trust_node.h"
+#include "smt/env_obj.h"
+#include "theory/theory.h"
+#include "theory/valuation.h"
+
+namespace cvc5 {
+
+class TheoryEngine;
+
+namespace prop {
+class PropEngine;
+}
+
+namespace theory {
+
+class PartitionGenerator : protected EnvObj
+{
+ public:
+  // PartitionGenerator
+  PartitionGenerator(Env& env,
+                     TheoryEngine* theoryEngine,
+                     prop::PropEngine* propEngine);
+
+  /**
+   * Make partitions for parallel solving. isFromFullCheck communicates whether
+   * makePartitions was call from the theory engine at a full check of a
+   * standard check.
+   */
+  TrustNode makePartitions(Theory::Effort e);
+
+ private:
+  /**
+   * Print the cube to the specified output.
+   */
+  void emitCube(Node toEmit);
+
+  /**
+   * Partition using the "revised" strategy, which emits cubes such as C1, C2,
+   * C3, !C1 & !C2 & !C3
+   */
+  TrustNode makeRevisedPartitions();
+
+  /**
+   * Block a path in the search by sending the not of toBlock as a lemma to the
+   * SAT solver.
+   */
+  TrustNode blockPath(TNode toBlock);
+
+  /**
+   * Stop partitioning and return unsat.
+   */
+  TrustNode stopPartitioning();
```

Comment:
It looks like this method can be `const`:

```suggestion
  TrustNode stopPartitioning() const;
```